test/integration/controlplane/transformation/transformation_testcase.go

```diff
@@ -218,27 +213,31 @@ func (e *transformTest) benchmark(b *testing.B) {
 }
 
 func (e *transformTest) getETCDPath() string {
+	return fmt.Sprintf("/%s/secrets/%s/%s", e.storageConfig.Prefix, e.ns.Name, e.secret.Name)
+}
+
+func (e *transformTest) getETCDPathForResource(group, resource, name, namespaceName string) string {
 	if e.secret != nil {
-		e.group = ""
-		e.resource = "secrets"
-		e.name = testSecret
-		e.namespaceName = testNamespace
+		group = ""
+		resource = "secrets"
+		name = testSecret
+		namespaceName = testNamespace
 	}
-	groupResource := e.resource
-	if e.group != "" {
-		groupResource = fmt.Sprintf("%s/%s", e.group, e.resource)
+	groupResource := resource
+	if group != "" {
+		groupResource = fmt.Sprintf("%s/%s", group, resource)
 	}
-	if e.namespaceName == "" {
-		return fmt.Sprintf("/%s/%s/%s", e.storageConfig.Prefix, groupResource, e.name)
+	if namespaceName == "" {
+		return fmt.Sprintf("/%s/%s/%s", e.storageConfig.Prefix, groupResource, name)
 	}
-	return fmt.Sprintf("/%s/%s/%s/%s", e.storageConfig.Prefix, groupResource, e.namespaceName, e.name)
+	return fmt.Sprintf("/%s/%s/%s/%s", e.storageConfig.Prefix, groupResource, namespaceName, name)
 }
 
 func (e *transformTest) getRawSecretFromETCD() ([]byte, error) {
-	etcdPath := e.getETCDPath()
-	etcdResponse, err := e.readRawRecordFromETCD(etcdPath)
+	secretETCDPath := e.getETCDPath()
+	etcdResponse, err := e.readRawRecordFromETCD(secretETCDPath)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read %s from etcd: %v", etcdPath, err)
+		return nil, fmt.Errorf("failed to read %s from etcd: %v", secretETCDPath, err)
 	}
 	return etcdResponse.Kvs[0].Value, nil
 }
```

Comment:
I reverted changes to `getRawSecretFromETCD` and `getETCDPath` and made those follow the same pattern as `run` - i.e. those are specific to `secrets` whereas the `runResource` and `getETCDPathForResource` are the generic ones.